elfpy/data/contract_interface.py

```diff
@@ -1,84 +1,55 @@
-"""Script to pull on-chain transaction data and output JSON for post-processing"""
+"""Functions and classes for interfacing with Hyperdrive smart contracts"""
 from __future__ import annotations
 
 import json
-import os
 import time
-from json import JSONEncoder
+import logging
 from typing import Iterable
 
+import attr
 import requests
 import toml
-from eth_utils.address import to_checksum_address
+from eth_typing import URI
 from hexbytes import HexBytes
 from web3 import Web3
-from web3.contract.contract import Contract, ContractEvent
-from web3.datastructures import AttributeDict, MutableAttributeDict
+from web3.contract.contract import ContractFunction, Contract, ContractEvent
 from web3.middleware import geth_poa
 from web3.types import ABIEvent, EventData, LogReceipt, TxReceipt
 
-# python `open` will infer the encoding if we do not specified, which is the behavior we want for now
-# pylint: disable=unspecified-encoding
+from elfpy.utils import apeworx_integrations as ape_utils
 
-# pylint: disable=too-many-locals
 
+@attr.s
+class HyperdriveAddressesJson:
+    """Addresses for deployed Hyperdrive contracts."""
 
-class ExtendedJSONEncoder(JSONEncoder):
-    """Overrides json encoder to handle hex inputs"""
+    # pylint: disable=too-few-public-methods
 
-    def default(self, o):
-        if isinstance(o, HexBytes):
-            return o.hex()
-        if isinstance(o, (AttributeDict, MutableAttributeDict)):
-            return dict(o)
-        return super().default(o)
+    hyperdrive: str = attr.ib()
+    base_token: str = attr.ib()
```

Comment:
believe that's a hyperdrivemath address deployed in the latest

Reply:
I'm not sure what you're saying. The full diff is super confusing because it tried to write contract_interface as a diff from acquire_transactions. This is the class:
```
@attr.s
class HyperdriveAddressesJson:
    """Addresses for deployed Hyperdrive contracts."""

    # pylint: disable=too-few-public-methods

    hyperdrive: str = attr.ib()
    base_token: str = attr.ib()
```